src/devices/machine/myb3k_kbd.cpp

```diff
@@ -140,16 +140,21 @@
 // for reference in the keyboard layout images.
 
 INPUT_PORTS_START( myb3k_keyboard )
+	PORT_START("MYB3K_CFG")
+	PORT_CONFNAME( 0x0003U, 0x0000U, "Keyboard Layout" )
+	PORT_CONFSETTING(       0x0000U, "Swedish Step/One" )
+	PORT_CONFSETTING(       0x0001U, "Australian JB3000" )
+	PORT_CONFSETTING(       0x0002U, "Japanese MyBrain3000" )
```

Comment:
If the keyboard is external and interchangeable, it should be a slot device allowing the different layouts to be plugged in.  If it's internal, it should instantiate the correct layout device for the machine.  Either way, these should be derived device classes, not a fake config port.

Reply:
Thanks cuavas, I think I understand. The keyboard is external and is electrically interchangeable, but it behaves as an internal one since the keyboard layout is tied to the driver/machine. Ie the swedish keyboard only works for stepone, the australian for jb3000, and the japanese for myb3k.

Thus I should change myb3k driver to initiate the  myb3k_keyboard_device accordingly. Drop the CONFNAME that can be manually changed and somehow configure the keyboard programmatically. 

Should I still use PORT_CONDITION for this, or is there a different way?

Have I understood correctly?

Reply:
You don't need `PORT_CONDITION` at all.  Just make derived classes that override the method returning the input port creator.  Use `PORT_INCLUDE` and `PORT_MODIFY` to modify the inputs.  See e.g. the Sun or Amiga keyboard matrixes.

Reply:
Fixed